server/src/test/java/org/elasticsearch/action/bulk/BulkProcessorTests.java

```diff
@@ -79,6 +79,7 @@ public void testBulkProcessorFlushPreservesContext() throws InterruptedException
             ThreadContext threadContext = threadPool.getThreadContext();
             assertEquals(headerValue, threadContext.getHeader(headerKey));
             assertSame(transientValue, threadContext.getTransient(transientKey));
+            listener.onResponse(new BulkResponse(new BulkItemResponse[0], 1L));
```

Comment:
I'm curious why this becomes necessary (I'm not pushing back on it, I'm just interested).

Reply:
In this PR I'm not actually asserting that the BulkRequest objects are closed (that comes in the next PR). But without this line the listener is never called so the BulkRequest is never called. Here's the stack where it closes the BulkRequest through this line (in a slightly different branch of code that will be in a future PR)

```
	at org.elasticsearch.action.bulk.BulkRequest.close(BulkRequest.java:503)
	at org.elasticsearch.core.IOUtils.close(IOUtils.java:71)
	at org.elasticsearch.core.Releasables.close(Releasables.java:34)
	at org.elasticsearch.core.Releasables.closeExpectNoException(Releasables.java:58)
	at org.elasticsearch.action.ActionListenerImplementations$2.run(ActionListenerImplementations.java:49)
	at org.elasticsearch.action.ActionListenerImplementations$RunAfterActionListener.onResponse(ActionListenerImplementations.java:262)
	at org.elasticsearch.action.ActionListener$3.onResponse(ActionListener.java:312)
	at org.elasticsearch.action.bulk.Retry$RetryHandler.finishHim(Retry.java:191)
	at org.elasticsearch.action.bulk.Retry$RetryHandler.onResponse(Retry.java:108)
	at org.elasticsearch.action.bulk.Retry$RetryHandler.onResponse(Retry.java:74)
	at org.elasticsearch.action.bulk.BulkProcessorTests.lambda$testBulkProcessorFlushPreservesContext$0(BulkProcessorTests.java:82)
	at org.elasticsearch.action.bulk.Retry$RetryHandler.execute(Retry.java:234)
	at org.elasticsearch.action.bulk.Retry$RetryHandler.execute(Retry.java:223)
	at org.elasticsearch.action.bulk.Retry.withBackoff(Retry.java:54)
	at org.elasticsearch.action.bulk.BulkRequestHandler.execute(BulkRequestHandler.java:59)
	at org.elasticsearch.action.bulk.BulkProcessor.execute(BulkProcessor.java:486)
	at org.elasticsearch.action.bulk.BulkProcessor.execute(BulkProcessor.java:496)
	at org.elasticsearch.action.bulk.BulkProcessor$Flush.run(BulkProcessor.java:537)
	at org.elasticsearch.threadpool.Scheduler$ReschedulingRunnable.doRun(Scheduler.java:223)
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at org.elasticsearch.threadpool.ThreadPool$1.run(ThreadPool.java:484)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
```